server/src/main/java/io/druid/client/cache/MapCache.java

```diff
@@ -17,59 +17,77 @@
 
 package io.druid.client.cache;
 
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.Weigher;
 import com.google.common.collect.Lists;
 import com.google.common.collect.Maps;
-import com.google.common.primitives.Ints;
 import com.metamx.emitter.service.ServiceEmitter;
 
 import java.nio.ByteBuffer;
-import java.util.Collections;
-import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.atomic.AtomicInteger;
-import java.util.concurrent.atomic.AtomicLong;
 
 /**
  */
 public class MapCache implements Cache
 {
-  public static Cache create(long sizeInBytes)
+  public static MapCache create(long sizeInBytes) {
+    return create(sizeInBytes, 0, 4);
+  }
+  public static MapCache create(long sizeInBytes, int initialCapacity, int concurrencyLevel)
   {
-    return new MapCache(new ByteCountingLRUMap(sizeInBytes));
+    return new MapCache(sizeInBytes, initialCapacity, concurrencyLevel);
   }
 
-  private final Map<ByteBuffer, byte[]> baseMap;
-  private final ByteCountingLRUMap byteCountingLRUMap;
-
-  private final Map<String, byte[]> namespaceId;
-  private final AtomicInteger ids;
-
-  private final Object clearLock = new Object();
+  private final com.google.common.cache.Cache<ByteBuffer, byte[]> baseCache;
+  private final com.google.common.cache.LoadingCache<String, Integer> namespaceId;
+  private final AtomicInteger ids = new AtomicInteger();
 
-  private final AtomicLong hitCount = new AtomicLong(0);
-  private final AtomicLong missCount = new AtomicLong(0);
-
-  MapCache(
-      ByteCountingLRUMap byteCountingLRUMap
-  )
+  private MapCache(long sizeInBytes, int initialCapacity, int concurrencyLevel)
   {
-    this.byteCountingLRUMap = byteCountingLRUMap;
-    this.baseMap = Collections.synchronizedMap(byteCountingLRUMap);
-
-    namespaceId = Maps.newHashMap();
-    ids = new AtomicInteger();
+    this.baseCache = CacheBuilder
+        .newBuilder()
+        .initialCapacity(initialCapacity)
+        .maximumWeight(sizeInBytes)
+        .recordStats()
+        .concurrencyLevel(concurrencyLevel)
+        .weigher(
+            new Weigher<ByteBuffer, byte[]>()
+            {
+              @Override
+              public int weigh(ByteBuffer key, byte[] value)
+              {
+                return key.remaining() + value.length;
+              }
+            }
+        )
+        .build();
+
+    this.namespaceId = CacheBuilder
```

Comment:
if my understanding is correct, there is no eviction on this cache, if so, i don't see why not using a simple concurrent hash map ?